### PR TITLE
fix: #5362, Password: (accessiblility) cannot access toggle button using keyboard

### PR DESCRIPTION
**Pull request by akshayaqburst**
###Defect Fixes
fix: #5362, Password: (accessiblility) cannot access toggle button using keyboard

### CODE REVIEW

File: components/lib/password/Password.vue [hunk 1]
```diff
@@ -277,6 +277,17 @@ export default {
         onMaskToggle() {
             this.unmasked = !this.unmasked;
         },
+        onToggleMaskKeyDown(event) {
+            switch (event.code) {
+                case 'Enter':
```
Comment (melloware) [inline components/lib/password/Password.vue:282]: I think you need `NumpadEnter` here too since you are using `code`

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/password/Password.vue (hunk 1 above)
Comment (akshayaqburst) [reply components/lib/password/Password.vue:282]: Done @melloware

**Comment by melloware** [comment]
It probably needs ARIA Labels too since its focusable.

File: components/lib/config/PrimeVue.js [hunk 2]
```diff
@@ -71,6 +71,8 @@ export const defaultOptions = {
         emptySelectionMessage: 'No selected item',
         emptySearchMessage: 'No results found',
         emptyMessage: 'No available options',
+        passwordHide: 'Hide Password',
+        passwordShow: 'Show Password',
```
Comment (melloware) [inline components/lib/config/PrimeVue.js:75]: Move this down into the `aria: {` section these are ARIA labels.

File: components/lib/password/Password.vue [hunk 3]
```diff
@@ -27,10 +27,10 @@
             :unstyled="unstyled"
         />
         <slot v-if="toggleMask && unmasked" name="hideicon" :onClick="onMaskToggle" :toggleCallback="onMaskToggle">
-            <component :is="hideIcon ? 'i' : 'EyeSlashIcon'" :class="[cx('hideIcon'), hideIcon]" @click="onMaskToggle" v-bind="ptm('hideIcon')" @keypress="onToggleMaskKeyDown" tabindex="0" />
+            <component :is="hideIcon ? 'i' : 'EyeSlashIcon'" :class="[cx('hideIcon'), hideIcon]" @click="onMaskToggle" v-bind="ptm('hideIcon')" @keypress="onToggleMaskKeyDown" tabindex="0" :aria-label="$primevue.config.locale.passwordHide" />
```
Comment (melloware) [inline components/lib/password/Password.vue:30]:
```suggestion
            <component :is="hideIcon ? 'i' : 'EyeSlashIcon'" :class="[cx('hideIcon'), hideIcon]" @click="onMaskToggle" v-bind="ptm('hideIcon')" @keypress="onToggleMaskKeyDown" tabindex="0" :aria-label="$primevue.config.locale.aria.passwordHide" />
```

File: components/lib/password/Password.vue [hunk 4]
```diff
@@ -27,10 +27,10 @@
...
         </slot>
         <slot v-if="toggleMask && !unmasked" name="showicon" :onClick="onMaskToggle" :toggleCallback="onMaskToggle">
-            <component :is="showIcon ? 'i' : 'EyeIcon'" :class="[cx('showIcon'), showIcon]" @click="onMaskToggle" v-bind="ptm('showIcon')" @keypress="onToggleMaskKeyDown" tabindex="0" />
+            <component :is="showIcon ? 'i' : 'EyeIcon'" :class="[cx('showIcon'), showIcon]" @click="onMaskToggle" v-bind="ptm('showIcon')" @keypress="onToggleMaskKeyDown" tabindex="0" :aria-label="$primevue.config.locale.passwordShow" />
```
Comment (melloware) [inline components/lib/password/Password.vue:33]:
```suggestion
            <component :is="showIcon ? 'i' : 'EyeIcon'" :class="[cx('showIcon'), showIcon]" @click="onMaskToggle" v-bind="ptm('showIcon')" @keypress="onToggleMaskKeyDown" tabindex="0" :aria-label="$primevue.config.locale.aria.passwordShow" />
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/password/Password.vue (hunk 3 above)
Comment (akshayaqburst) [reply components/lib/password/Password.vue]: Done

File: components/lib/config/PrimeVue.js (hunk 2 above)
Comment (akshayaqburst) [reply components/lib/config/PrimeVue.js]: Done

**Review by melloware — APPROVED** [review]

**Comment by tugcekucukoglu** [comment]
https://github.com/primefaces/primevue/issues/5362#issuecomment-1994359529

**Pull request closed without merging**